common/scala/src/whisk/core/entity/WhiskTrigger.scala

```diff
@@ -60,12 +100,28 @@ case class WhiskTrigger(
     limits: TriggerLimits = TriggerLimits(),
     version: SemVer = SemVer(),
     publish: Boolean = false,
-    annotations: Parameters = Parameters())
+    annotations: Parameters = Parameters(),
+    rules: Option[Map[Namespace, ReducedRule]] = Some(Map[Namespace, ReducedRule]()))
```

Comment:
We prefer saving an empty rule instead of a None. This has the advantage, that `addRule`and `removeRule` are much easier to realize.
The next advantage is, that we have a map as value and we only have to handle if it is empty or not.
The only reason, why `rules` is an Option is, to tolerate the current database schema.